core/src/main/java/com/datastrato/gravitino/listener/event/Event.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.listener.event;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.utils.PrincipalUtils;
+
+/** The base class for all event, contains user and resource identifier. */
+@Evolving
+public abstract class Event {
+
+  private UserAndIdentifierInfo userAndIdentifierInfo;
+
+  /**
+   * create a Event instance.
+   *
+   * @param userAndIdentifierInfo The user and resource identifier information.
+   */
+  public Event(UserAndIdentifierInfo userAndIdentifierInfo) {
```

Comment:
Event provides a public method `getUserAndIdentifierInfo` to get user and identifier information,  seems no `package private` do I miss something?

Reply:
I mean you should control the access permission for your classes and methods to users. Simply for event, user will only use it, it will never create it or inherit it, so provide limited access permissions.

Reply:
ok